nested_lookup/lookup_api.py

```diff
@@ -29,28 +31,166 @@ def _nested_delete(document, key):
     return document
 
 
-def nested_update(document, key, value, in_place=False):
+def nested_update(document: dict, key: str, value: object,
+                  in_place: bool = False,
+                  treat_as_element: bool = True) -> dict:
+    """
```

Comment:
Same as the remark regarding `nested_delete` a document could either a `list` or a `dict` therefore the return type could be either a `list` or a `dict`.